### ISSUE TITLE
Add support for vAlign in table cells

### ISSUE THREAD

**Issue by tripodsan** [mwilliamson/mammoth.js#338]
the vertical alignment of a table cell is not recognized:

eg:

```xml
<w:tc>
    <w:tcPr>
        <w:tcW w:w="2277" w:type="dxa"/>
        <w:vAlign w:val="bottom"/>
    </w:tcPr>
    <w:p w14:paraId="0CEE5408" w14:textId="426F30F1" w:rsidR="00C348D8"
         w:rsidRDefault="00C348D8" w:rsidP="00C348D8">
        <w:pPr>
            <w:jc w:val="center"/>
        </w:pPr>
        <w:r>
            <w:t>Bottom center</w:t>
        </w:r>
    </w:p>
</w:tc>
```

it would be easy to add to the `readTableCell()` function.

PR follows...

**Comment by mwilliamson** [comment]
Mammoth intentionally doesn't try to copy formatting verbatim, so adding something along the lines of what was in the pull request (a `vAlign` attribute to `td` elements) doesn't seem like the right approach to me.

**Comment by mwilliamson** [comment]
It would also be useful to have a minimal example document, the expected HTML, and the actual HTML when converting that document.

**Comment by tripodsan** [comment]
> Mammoth intentionally doesn't try to copy formatting verbatim

we could narrow it down to the vAlign values supported by html... the PR just detects `center` and converts it to `middle`.
all others: `top`, `bottom`, `baseline` are identical.

> It would also be useful to have a minimal example document, the expected HTML, and the actual HTML when converting that document.

this was covered with the test in the PR :-)

**Comment by mwilliamson** [comment]
> > Mammoth intentionally doesn't try to copy formatting verbatim
> 
> we could narrow it down to the vAlign values supported by html... the PR just detects `center` and converts it to `middle`. all others: `top`, `bottom`, `baseline` are identical.

This is still trying to copy the formatting from the source document into the HTML, which in general is not the approach that Mammoth takes.

**Comment by tripodsan** [comment]
> This is still trying to copy the formatting from the source document into the HTML, which in general is not the approach that Mammoth takes.

but [other formats](https://github.com/mwilliamson/mammoth.js/blob/9c273d07a3f5970033ed53d6e108b6f5ab3fd5eb/lib/documents.js#L58-L64), for example the `verticalAlignment` on text runs,  are also _copied_.

so how would you expose the formatting of the table cells then?

**Comment by mwilliamson** [comment]
> > This is still trying to copy the formatting from the source document into the HTML, which in general is not the approach that Mammoth takes.
> 
> but [other formats](https://github.com/mwilliamson/mammoth.js/blob/9c273d07a3f5970033ed53d6e108b6f5ab3fd5eb/lib/documents.js#L58-L64), for example the `verticalAlignment` on text runs, are also _copied_.

In the case of vertical alignment, we infer some semantic meaning: if the vertical alignment is (for instance) subscript, then we can convert that into a `sub` element in the outputted HTML.

> so how would you expose the formatting of the table cells then?

I'm not sure! I haven't had time to think about it that much.

**Comment by tripodsan** [comment]
>  we infer some semantic meaning

fair enough... but `isAllCaps` is purely formatting :-)

**Comment by mwilliamson** [comment]
> > we infer some semantic meaning
> 
> fair enough... but `isAllCaps` is purely formatting :-)

Indeed: and that's why Mammoth, by default, doesn't change the outputted HTML when a run is all caps.
